lib/application.js

```diff
@@ -208,7 +208,7 @@ module.exports = class Application extends Emitter {
     request.response = response
     response.request = request
     context.originalUrl = request.originalUrl = req.url
-    context.state = {}
+    context.state = Object.assign({}, context.state)
```

Comment:
The whole point of `ctx.state` as I see it is to ensure there's no state bleed between requests.
As far as I see it, this change defeats this original purpose.